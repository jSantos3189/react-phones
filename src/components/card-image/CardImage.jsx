import React from 'react';
import PropTypes from 'prop-types';
import './cardImage.css';

const CardImage = ({ bg, title }) => (
  <div className="CardImage" style={{ backgroundImage: `url(${bg})` }}>
    <h1 className="CardImage-title">{title}</h1>
  </div>
);

CardImage.propTypes = {
  bg: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default CardImage;
