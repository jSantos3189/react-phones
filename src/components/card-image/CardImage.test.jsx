import React from 'react';
import { shallow } from 'enzyme';
import CardImage from './CardImage';

describe('CardImageComponent', () => {
  let cardImageComponent;

  beforeEach(() => {
    cardImageComponent = shallow(
      <CardImage bg="path/image" title="react-phonebook" />,
    );
  });

  it('should render <CardImage />', () => {
    expect(cardImageComponent.getElements()).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(cardImageComponent.find('h1')).toHaveLength(1);
  });

  it('should have a background-image', () => {
    expect(cardImageComponent.hasClass('CardImage')).toBe(true);
  });
});
