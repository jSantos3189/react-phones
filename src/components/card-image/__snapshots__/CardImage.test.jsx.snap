// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImageComponent should render <CardImage /> 1`] = `
Array [
  <div
    className="CardImage"
    style={
        Object {
            "backgroundImage": "url(path/image)",
          }
    }
>
    <h1
        className="CardImage-title"
    >
        react-phonebook
    </h1>
</div>,
]
`;
